Segfault during CI run

Observed fairly frequently, e.g. [here](https://github.com/CQCL/pytket-quantinuum/actions/runs/16318040670/job/46088586136?pr=616):
```
============================= test session starts ==============================
platform linux -- Python 3.11.13, pytest-8.4.1, pluggy-1.6.0
rootdir: /home/runner/work/pytket-quantinuum/pytket-quantinuum
configfile: pytest.ini
plugins: rerunfailures-15.1, timeout-2.4.0, requests-mock-1.12.1, hypothesis-6.135.32
collected 155 items

integration/backend_test.py ..............ssss..........x+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
~~~~~~~~~~~~~~~~~~~~~ Stack of asyncio_0 (139879195342528) ~~~~~~~~~~~~~~~~~~~~~
  File "/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/threading.py", line 1002, in _bootstrap
    self._bootstrap_inner()
  File "/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/concurrent/futures/thread.py", line 81, in _worker
    work_item = work_queue.get(block=True)
+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
x................................sss..................s.s............... [ 65%]
..............s...                                                       [ 76%]
integration/local_emulator_multithreading_test.py ss                     [ 78%]
/home/runner/work/_temp/acee1b5b-42d2-436f-aa9a-490e76ed2c10.sh: line 1:  2159 Segmentation fault      (core dumped) pytest integration/
integration/local_emulator_test.py ssssssssssssssssssssssssssssssssss    [100%]
Error: Process completed with exit code 139.
```

--- a/tests/integration/backend_test.py
+++ b/tests/integration/backend_test.py
@@ -13,7 +13,6 @@
 # limitations under the License.
 
 import datetime
-import gc
 import json
 import os
 import time
@@ -77,6 +76,13 @@ REASON_PROD = "PYTKET_RUN_REMOTE_TESTS_PROD not set \
 
 REASON_MPL = "PYTKET_RUN_MPL_TESTS not set \
 (requires configuration of Quantinuum username)"
+
+
+def prog_from_qir_text(qir: str) -> str:
+    ctx = create_context()
+    module = parse_assembly(qir, context=ctx)
+    ir = module.as_bitcode()
+    return b64encode(ir).decode("utf-8")
 
 
 @pytest.mark.skipif(skip_remote_tests, reason=REASON)
@@ -180,7 +186,6 @@ def test_bell(
 def test_multireg(
     authenticated_quum_backend_qa: QuantinuumBackend, language: Language
 ) -> None:
-    gc.disable()
     b = authenticated_quum_backend_qa
     c = Circuit()
     q1 = Qubit("q1", 0)
@@ -976,18 +981,12 @@ def test_tk2(
 )
 @pytest.mark.timeout(120)
 def test_qir_submission(authenticated_quum_backend_qa: QuantinuumBackend) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_qa
 
     with open("integration/qir/qat-link_2.ll") as f:
         qir = f.read()
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
-    h = b.submit_program(Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10)
+    h = b.submit_program(Language.QIR, prog_from_qir_text(qir), n_shots=10)
     r = b.get_result(h)
     assert set(r.get_bitlist()) == set([Bit("0_t0", 0), Bit("0_t1", 0)])  # noqa: C405
     assert len(r.get_shots()) == 10
@@ -999,18 +998,12 @@ def test_qir_submission(authenticated_quum_backend_qa: QuantinuumBackend) -> Non
 )
 @pytest.mark.timeout(120)
 def test_qir_entrypoints(authenticated_quum_backend_prod: QuantinuumBackend) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_prod
 
     with open("integration/qir/qat-link.ll") as f:
         qir = f.read()
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
-    h = b.submit_program(Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10)
+    h = b.submit_program(Language.QIR, prog_from_qir_text(qir), n_shots=10)
     r = b.get_result(h)
     assert set(r.get_bitlist()) == set([Bit("0_t0", 0), Bit("0_t1", 0)])  # noqa: C405
     assert len(r.get_shots()) == 10
@@ -1022,18 +1015,12 @@ def test_qir_entrypoints(authenticated_quum_backend_prod: QuantinuumBackend) -> 
 )
 @pytest.mark.timeout(120)
 def test_qir_entrypoints_qa(authenticated_quum_backend_qa: QuantinuumBackend) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_qa
 
     with open("integration/qir/qat-link.ll") as f:
         qir = f.read()
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
-    h = b.submit_program(Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10)
+    h = b.submit_program(Language.QIR, prog_from_qir_text(qir), n_shots=10)
     r = b.get_result(h)
     assert set(r.get_bitlist()) == set([Bit("0_t0", 0), Bit("0_t1", 0)])  # noqa: C405
     assert len(r.get_shots()) == 10
@@ -1047,17 +1034,11 @@ def test_qir_entrypoints_qa(authenticated_quum_backend_qa: QuantinuumBackend) ->
 def test_qir_submission_mz_to_reg(
     authenticated_quum_backend_prod: QuantinuumBackend,
 ) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_prod
     with open("integration/qir/test_pytket_qir_6.ll") as f:
         qir = f.read()
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
-    h = b.submit_program(Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10)
+    h = b.submit_program(Language.QIR, prog_from_qir_text(qir), n_shots=10)
 
     r = b.get_result(h)
 
@@ -1073,17 +1054,11 @@ def test_qir_submission_mz_to_reg(
 def test_qir_submission_mz_to_reg_qa(
     authenticated_quum_backend_qa: QuantinuumBackend,
 ) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_qa
     with open("integration/qir/test_pytket_qir_6.ll") as f:
         qir = f.read()
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
-    h = b.submit_program(Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10)
+    h = b.submit_program(Language.QIR, prog_from_qir_text(qir), n_shots=10)
     r = b.get_result(h)
     assert len(r.get_shots()) == 10
     assert len(r.get_bitlist()) == 20
@@ -1104,20 +1079,14 @@ def test_qir_submission_mz_to_reg_qa(
 def test_qir_submission_64bitwasm5_qa(
     authenticated_quum_backend_qa: QuantinuumBackend, filename: str
 ) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_qa
     with open(f"integration/qir/{filename}") as f:
         qir = f.read()
 
     wfh = WasmFileHandler(str(Path(__file__).parent.parent / "wasm" / "add1.wasm"))
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
     h = b.submit_program(
-        Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10, wasm_file_handler=wfh
+        Language.QIR, prog_from_qir_text(qir), n_shots=10, wasm_file_handler=wfh
     )
     r = b.get_result(h)
     assert len(r.get_shots()) == 10
@@ -1140,20 +1109,14 @@ def test_qir_submission_64bitwasm5_qa(
 def test_qir_submission_64bitwasm6_qa(
     authenticated_quum_backend_qa: QuantinuumBackend, filename: str
 ) -> None:
-    # disable Garbage Collector because of
-    # https://github.com/CQCL/pytket-quantinuum/issues/170
-    gc.disable()
     b = authenticated_quum_backend_qa
     with open(f"integration/qir/{filename}") as f:
         qir = f.read()
 
     wfh = WasmFileHandler(str(Path(__file__).parent.parent / "wasm" / "add1.wasm"))
 
-    ctx = create_context()
-    module = parse_assembly(qir, context=ctx)
-    ir = module.as_bitcode()
     h = b.submit_program(
-        Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10, wasm_file_handler=wfh
+        Language.QIR, prog_from_qir_text(qir), n_shots=10, wasm_file_handler=wfh
     )
     with pytest.raises(ValueError):
         # this circuit gives a negative results which we reject

